### DISCUSSION TITLE
Running the server in background Perpetually

### DISCUSSION THREAD

**Discussion by Curiosity007** [tecoholic/ner-annotator#83] [Q&A] [answered]
Main objective is to keep the server running. But "nohup yarn serve &" is not achieving this result. Any alternatives?

**Comment by tecoholic** [comment, marked as answer]
Not sure what your usecase is, the alternatives would be to:

1. Build the desktop app for your platform and install it
2. Build the app into static website and deploy it using something like Nginx, like https://tecoholic.github.io/ner-annotator/

**Comment by Curiosity007** [comment]
I am using Linux OCI to host this. Now I have limited knowledge about hosting/configuring server. What I was doing, ran the code (nohup yarn serve &) via putty, and accessing the url, which has the same UI as https://tecoholic.github.io/ner-annotator/ . 

Can you kindly point me to the right direction on what needs to be done.

**Reply by tecoholic** [reply]
Use the [npm run build](https://github.com/tecoholic/ner-annotator/blob/main/package.json#L7) command to build the app into a static site. It should create a index.html and all the associated files in the "build" directory. Copy its contents to the root of your web server - this would be whatever is in the Linux OCI (I have no knowledge about it). Usually it is either Apache or Nginx. Check port 80 to see if there is something like that already running. If not, install either of them and put the files in the default site directory. Google is your friend on this :)

**Reply by Curiosity007** [reply]
Thanks. Got it running. 

1. Installed Nginx 
2. Ran "npm run build"
3. Copied the contents of "docs" folder into /var/www/html/ner-annotator (Creating this folder is important. Because somehow the website is looking for js files inside this folder only)

Ran the website and working perfectly. Thanks
